integrations/aws/main.py

```diff
@@ -0,0 +1,184 @@
+import json
+from typing import Any
+
+import aioboto3
+from port_ocean.core.models import Entity
+from utils import ACCOUNT_ID_PROPERTY, KIND_PROPERTY, REGION_PROPERTY, ResourceKindsWithSpecialHandling, describe_accessible_accounts, batch_resources, describe_single_resource, _fix_unserializable_date_properties, _get_sessions, find_account_id_by_session, get_resource_kinds_from_config, is_global_resource, update_available_access_credentials, validate_request, get_matching_kinds_from_config
+from port_ocean.context.ocean import ocean
+from loguru import logger
+from starlette.requests import Request
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+DEFAULT_REGION = "us-east-1"
+
+@ocean.on_resync()
+async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    await update_available_access_credentials()
```

Comment:
why is this needed?

Reply:
This is the refresh of the roles credentials since they need to be refreshed every now and then